catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java

```diff
@@ -0,0 +1,502 @@
+        }
+
+        return createResponse;
+    }
+
+    /**
+     * Version updated metacards based off of the successfully created metacards found in the
+     * {@link UpdateResponse}
+     *
+     * @param updateResponse Versioned metacards created from any created metacards in this
+     * @return The original UpdateResponse
+     * @throws SourceUnavailableException
+     * @throws IngestException
+     */
+    public UpdateResponse versionUpdate(UpdateResponse updateResponse)
+            throws SourceUnavailableException, IngestException {
+        if (doSkip(updateResponse)) {
+            return updateResponse;
+        }
+        setSkipFlag(updateResponse);
+
+        List<Metacard> inputMetacards = updateResponse.getUpdatedMetacards()
+                .stream()
+                .map(Update::getNewMetacard)
+                .collect(Collectors.toList());
+
+        CreateResponse createdHistory = getVersionedMetacards(inputMetacards,
+                MetacardVersion.Action.UPDATED);
+        if (createdHistory == null) {
+            return updateResponse;
+        }
+
+        if (updateResponse.getProperties() != null) {
+            updateResponse.getProperties()
+                    .put(HISTORY_METACARDS_PROPERTY,
+                            new ArrayList<>(createdHistory.getCreatedMetacards()));
+        }
+
+        return updateResponse;
+    }
+
+    /**
+     * Versions created {@link Metacard}s and {@link ContentItem}s.
+     *
+     * @param createStorageRequest Versions this requests items
+     * @return The transaction key for using with {@link Historian#commit(String)} and
+     * {@link Historian#rollback(String)}
+     * @throws IngestException
+     * @throws StorageException
+     */
+    public Optional<String> versionStorageCreate(CreateStorageRequest createStorageRequest)
+            throws IngestException, StorageException {
+        if (doSkip(createStorageRequest)) {
+            return Optional.empty();
+        }
+        setSkipFlag(createStorageRequest);
+
+        String transactionKey = UUID.randomUUID()
+                .toString();
+        List<Callable<Boolean>> preStaged = new ArrayList<>(2);
+        staged.put(transactionKey, preStaged);
+
+        @SuppressWarnings("unchecked")
+        Map<String, Path> tmpContentPaths = (Map<String, Path>) createStorageRequest.getProperties()
+                .getOrDefault(CONTENT_PATHS, new HashMap<>());
+
+        List<ContentItem> newContentItems = createStorageRequest.getContentItems()
+                .stream()
+                .map(ContentItem::getMetacard)
+                .distinct()
+                .flatMap(mc -> getVersionedContent(mc,
+                        createStorageRequest.getContentItems(),
+                        MetacardVersion.Action.CREATED_CONTENT,
+                        tmpContentPaths).stream())
+                .collect(Collectors.toList());
+
+        newContentItems.forEach(ci -> {
```

Comment:
Yes and you made me catch a bug i forgot to filter out generated content items (like preview and overview)
done